airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -82,7 +119,271 @@ public JsonNode toJdbcConfig(JsonNode mssqlConfig) {
+              config.get("database").asText(), sql));
+          return ps;
+        }, JdbcUtils::rowToJson).collect(toList());
+
+        if (queryResponse.size() < 1) {
+          throw new RuntimeException(String.format(
+              "Couldn't find '%s' in sys.databases table. Please check the spelling and that the user has relevant permissions (see docs).",
+              config.get("database").asText()));
+        }
+
+        if (queryResponse.get(0).get("snapshot_isolation_state").asInt() != 1) {
+          throw new RuntimeException(String.format(
+              "Detected that snapshot isolation is not enabled for database '%s'. MSSQL CDC relies on snapshot isolation. "
+                  + "Please check the documentation on how to enable snapshot isolation on MS SQL Server.",
+              config.get("database").asText()));
+        }
+      });
+
+    }
+
+    return checkOperations;
+  }
+
+  @Override
+  public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(JsonNode config,
+                                                                             JdbcDatabase database,
+                                                                             ConfiguredAirbyteCatalog catalog,
+                                                                             Map<String, TableInfoInternal> tableNameToTable,
+                                                                             JdbcStateManager stateManager,
+                                                                             Instant emittedAt) {
+    if (isCdc(config) && shouldUseCDC(catalog)) {
+      LOGGER.info("using CDC: {}", true);
+      // TODO: Figure out how to set the isCDC of stateManager to true. Its always false
+      // State works differently in CDC than it does in convention incremental. The state is written to an
+      // offset file that debezium reads from. Then once all records are replicated, we read back that
+      // offset file (which will have been updated by debezium) and set it in the state. There is no
+      // incremental updating of the state structs in the CDC impl.
+      final AirbyteFileOffsetBackingStore offsetManager = initializeState(stateManager);
+      AirbyteSchemaHistoryStorage schemaHistoryManager = initializeDBHistory(stateManager);
+      FilteredFileDatabaseHistory.setDatabaseName(config.get("database").asText());
+
+      final Lsn targetLsn = getLsn(database);
+      LOGGER.info("identified target lsn: " + targetLsn);
+
+      /**
+       * We use 100,000 as capacity. We've used default * 10 queue size and batch size of debezium :
+       * {@link io.debezium.config.CommonConnectorConfig#DEFAULT_MAX_BATCH_SIZE} is 2048 (so 20,480)
+       * {@link io.debezium.config.CommonConnectorConfig#DEFAULT_MAX_QUEUE_SIZE} is 8192 (so 81,920)
+       */
+      final LinkedBlockingQueue<ChangeEvent<String, String>> queue = new LinkedBlockingQueue<>(100000);
```

Comment:
As discussed we should probably stick with the default numbers so queue should be of 10k size